MRN-Code/coinstac
# Update Test Comp Schema & Refactor GraphQL Field Sharing

* **Problem:**
   1. The computation schema has changed slightly
   1. GraphQL field and function names are inconsistent
   1. GraphQL field sharing still repetitive in some places
* **Solution**
   1. Update test computations' schema to match spec
   1. Further separate GraphQL fields to share not only functions but fields within functions
* **Tests:**
   1. Follow the steps outlined in `packages/coinstac-graphql/README.md` to setup your development environment with RethinkDB, start the GraphQL API, and insert a couple computations into the new database.
   1. Duplicate `packages/coinstac-graphql/config/api-config-sample.json` as `api-config.json`.
   1. Duplicate `packages/coinstac-graphql/config/comp-reg-config-sample.json` as `comp-reg-config.json` and uncomment the contents.
   1. Start the application and login with user: `test` & pass: `password` to test authentication.
   1. Notice that there is no route listed called "Submit Computation". That is because this user does not have write access.
   1. Refresh the application and login with a random user. An error should appear.
   1. Click on the Create User tab and create a new user using `test` as the username. You should see an error telling you that the username is already in use.
   1. Create a new random user to test that user creation works.
   1. Refresh the application and login with user: `author` & password: `password`. Notice that you have a navigation item called Submit Computation.
